### PR TITLE
added FCM

### LINKED ISSUE #25: Setup FCM Listners
- Firebase cloud messaging to send inapp notifications
- Adds notification icons and other configs/intents

### PR DESCRIPTION
**Pull request by jaynit1106**
# Description

added FCM to the project

Fixes #25 
some samples:-

https://user-images.githubusercontent.com/77606730/147570513-c0270b09-b943-4c22-99c2-8e6416af69b0.mp4

https://user-images.githubusercontent.com/77606730/147570533-15a6c6f7-7940-4fea-870b-4ade78df5029.mp4

### CODE REVIEW

File: customer_app/lib/services/graphql_services.dart [hunk 1]
```diff
@@ -1,16 +1,16 @@
-//import 'package:get/get.dart';
 import 'package:flutter/material.dart';
+import 'package:get/get_core/src/get_main.dart';
```
Comment (abhibhaw) [inline customer_app/lib/services/graphql_services.dart:2]: Again just importing get/get.dart is suffiecient Multiple imports should be avoided

**Review by abhibhaw — CHANGES_REQUESTED** [review]
Just minor changes, review once. Also please attach a few notification ss in this pr for reference

File: customer_app/lib/views/screens/home/customer_app.dart [hunk 2]
```diff
@@ -9,6 +10,19 @@ import 'package:customer_app/views/widgets/home/shared/bottom_nav_bar.dart';
...
+
+  void initState(){
+
+    final firebaseMessaging=FCM();
+    firebaseMessaging.setNotifications();
+    firebaseMessaging.streamCtrl.stream.listen(_changeData);
+    firebaseMessaging.titleCtrl.stream.listen(_changeTitle);
+    firebaseMessaging.bodyCtrl.stream.listen(_changeBody);
+  }
+  _changeData(String msg)=>print(msg);
```
Comment (abhibhaw) [inline customer_app/lib/views/screens/home/customer_app.dart:22]: Unnecessary print statement. Can be removed ig?

**Review by abhibhaw — CHANGES_REQUESTED** [review]

File: customer_app/lib/views/screens/home/customer_app.dart (hunk 2 above)
Comment (jaynit1106) [reply customer_app/lib/views/screens/home/customer_app.dart:22]: they are for foreground notification

**Comment by abhibhaw** [comment]
TODO: #29

**Review by abhibhaw — APPROVED** [review]
lgtm

**Comment by abhibhaw** [comment]
Also a suggestion, Please write better commit messages.

**Pull request merged**